File: phase-5-roblox/5.3-oop-in-luau/starter/roblox-kingdom/Engine/Lumberyard.lua
local Building = require(script.Parent.Building)

local Lumberyard = setmetatable({}, { __index = Building })
Lumberyard.__index = Lumberyard

function Lumberyard.new(name: string)
    local self = Building.new(name)
    setmetatable(self, Lumberyard)
    return self
end

function Lumberyard:tick(ledger: any)
    ledger:add("Wood", 3 * self.level)
end

return Lumberyard

File: phase-2-persistence/2.3-round-trip-tests/starter/Kingdom.Engine/Buildings/Lumberyard.cs
namespace Kingdom.Engine.Buildings;

public class Lumberyard : Building
{
    public Lumberyard(string name) : base(name) { }
    public Lumberyard(string name, int level) : base(name, level) { }

    public override void Tick(ResourceLedger ledger)
    {
        ledger.Add(Resource.Wood, 3 * Level);
    }
}
